### PR TITLE
PRMT-3523 ORC E2E Testing Remaining Dependancy Injection

### PR DESCRIPTION
**Pull request by joefong-nhs**
- Dependency injected for pdsAdaptorService and Gp2gpMessengerService
- For several tests, change from field injection to constructor injection

### CODE REVIEW

File: src/main/java/uk/nhs/prm/e2etests/model/nems/NemsResolutionMessage.java [hunk 1]
```diff
@@ -2,11 +2,25 @@
...
+
 @Getter
 public class NemsResolutionMessage {
     String nemsMessageId;
     String status;
 
+    public boolean hasTheSameContentAs(Object o) {
+        if (this == o) return true;
+        if (o == null) return false;
+        NemsResolutionMessage that = (NemsResolutionMessage) o;
```
Comment (MohammadIqbalAD-NHS) [inline src/main/java/uk/nhs/prm/e2etests/model/nems/NemsResolutionMessage.java:15]: What kind of variable name is 'that' !?

File: src/main/java/uk/nhs/prm/e2etests/utility/NhsIdentityGenerator.java [hunk 2]
```diff
@@ -32,10 +33,6 @@ public static String randomNemsMessageId() {
     }
 
     public static String randomNemsMessageId(boolean shouldLog) {
-        String nemsMessageId = UUID.randomUUID().toString();
-        if (shouldLog) {
-            System.out.println("Generated Nems Message ID " + nemsMessageId);
-        }
-        return nemsMessageId;
+        return UUID.randomUUID().toString();
```
Comment (joefong-nhs) [inline src/main/java/uk/nhs/prm/e2etests/utility/NhsIdentityGenerator.java:36]: the boolean `shouldLog` is unused now. We should either use it or remove it from arguments

File: src/main/java/uk/nhs/prm/e2etests/service/PdsAdaptorService.java [hunk 3]
```diff
@@ -1,40 +1,62 @@
...
+    public PdsAdaptorService(PdsAdaptorProperties pdsAdaptorProperties, @Value("${test.pds.username}") String username) {
+        this.patientRootUrl = pdsAdaptorProperties.getPdsAdaptorUrl();
         this.username = username;
-        this.apiKey = apiKey;
-        this.patientRootUrl = pdsAdaptorUrl;
+
+        switch (username) {
+            case "live-test" -> this.apiKey = pdsAdaptorProperties.getLiveTestApiKey();
+            case "e2e-test" -> this.apiKey = pdsAdaptorProperties.getE2eTestApiKey();
+            case "performance-test" -> this.apiKey = pdsAdaptorProperties.getPerformanceApiKey();
```
Comment (joefong-nhs) [inline src/main/java/uk/nhs/prm/e2etests/service/PdsAdaptorService.java:37]: Could we possibly replace the 3 strings here with Enum class?  (Although I'm not sure how)

File: src/main/java/uk/nhs/prm/e2etests/model/nems/NemsResolutionMessage.java (hunk 1 above)
Comment (martin-nhs) [reply src/main/java/uk/nhs/prm/e2etests/model/nems/NemsResolutionMessage.java:15]: Funnily enough, when you override equals and hashcode - this does seem to be the autogenerated variable name!

File: src/main/java/uk/nhs/prm/e2etests/utility/NhsIdentityGenerator.java (hunk 2 above)
Comment (martin-nhs) [reply src/main/java/uk/nhs/prm/e2etests/utility/NhsIdentityGenerator.java]: Addressed

File: src/main/java/uk/nhs/prm/e2etests/model/nems/NemsResolutionMessage.java (hunk 1 above)
Comment (martin-nhs) [reply src/main/java/uk/nhs/prm/e2etests/model/nems/NemsResolutionMessage.java:16]: Addressed

File: src/main/java/uk/nhs/prm/e2etests/service/PdsAdaptorService.java [hunk 4]
```diff
@@ -1,40 +1,62 @@
...
 
     private final String apiKey;
     private final String patientRootUrl;
     private final String username;
+
     private final RestTemplate restTemplate = new RestTemplate();
 
-    public PdsAdaptorService(String username, String apiKey, String pdsAdaptorUrl) {
+    @Autowired
+    public PdsAdaptorService(PdsAdaptorProperties pdsAdaptorProperties, @Value("${test.pds.username}") String username) {
```
[the remark this answers is not in the document]
Comment (martin-nhs) [reply src/main/java/uk/nhs/prm/e2etests/service/PdsAdaptorService.java:30]: I agree that `username` does not exactly fit to the value in which it holds and can be quite misleading, though I do believe that within the context of the `PdsAdaptorService` - it is still valid to be called username. The PDS service requires a 'username' and 'password' i.e. API key to be passed via a header, so technically would not want this to align with what it expects?

Please correct me if I'm wrong though! 😄

**Review by martin-nhs — APPROVED** [review]

**Pull request merged**